### PR TITLE
Iodine database

### PR DESCRIPTION
**Pull request by sarahkha**
This PR is for the thermodynamics and kinetic data related to iodine chemistry 
H-abstraction and R_recombination were added

### CODE REVIEW

**Comment by alongd** [comment]
@sarahkha, the tests for this branch complain that `Duplicate group name HCl found in kinetics groups for H_Abstraction/groups family.` Will you take a look?

File: input/kinetics/libraries/Iodine-R_recombination/reactions.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+shortDesc = u""" 
+(see Table S2 of Supplementary material of Buras et al.)
+longDesc = u"""
+
+entry(
+    index = 1,
+    label = "H + I + Ar <=> HI + Ar",
+    degeneracy = 1,
+    kinetics = Arrhenius(
+        A = (1.3E-31, 'cm^3/(molecule*s)'),
```
Comment (alongd) [inline input/kinetics/libraries/Iodine-R_recombination/reactions.py]: The units are for a 2nd-order reaction, but this looks like a 3rd-order one.
Same comment for the entry below

File: input/kinetics/libraries/Iodine-R_recombination/reactions.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    shortDesc = u"""Estimated: thermochemical, kinetic, or other""",
+    longDesc =
+u"""
+Bath gas: 
+Excitation technique: 
+Analytical technique: 
+""",
+)
+
+"""
```
Comment (alongd) [inline input/kinetics/libraries/Iodine-R_recombination/reactions.py]: This `"""` line causes everything below to be read as a comment, not as python code. Should be deleted.

File: input/kinetics/libraries/Iodine-R_recombination/reactions.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        pages = """2067-2076""",
+        year = "1982",
+        url = "http://pubs.rsc.org/-/content/articlehtml/1982/f2/f29827802067",
+    ),
+    referenceType = "experiment",
+    shortDesc = u"""Estimated from I + C2H5 -> C2H5I by Zach et al (Table S2)""",
+    longDesc =
+u"""
+Bath gas: Kr (100 Torr)
+Excitation technique:
```
Comment (alongd) [inline input/kinetics/libraries/Iodine-R_recombination/reactions.py]: missing information

File: input/kinetics/libraries/Iodine-R_recombination/reactions.py [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        pages="""2067-2076""",
+        year="1982",
+        url="http://pubs.rsc.org/-/content/articlehtml/1982/f2/f29827802067",
+    ),
+    referenceType="experiment",
+    shortDesc=u"""Estimated from I + C2H5 -> C2H5I by Zach et al (Table S2)""",
+    longDesc=
+    u"""
+    Bath gas: Kr (100 Torr)
+    Excitation technique:
```
Comment (alongd) [inline input/kinetics/libraries/Iodine-R_recombination/reactions.py]: missing info

File: input/kinetics/libraries/Iodine-R_recombination/reactions.py [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+        pages = """2409−2434""",
+        year = "2012",
+        url = "https://pubs.acs.org/doi/pdf/10.1021/jp207461a",
+    ),
+    referenceType = "estimate",
+    shortDesc = u"""Estimated: thermochemical, kinetic, or other""",
+    longDesc =
+u"""
+Bath gas: 
+Excitation technique:
```
Comment (alongd) [inline input/kinetics/libraries/Iodine-R_recombination/reactions.py]: missing info

File: input/kinetics/libraries/Iodine-R_recombination/reactions.py [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+        pages = """ """,
+        year = "1981",
+        url = "https://aip.scitation.org/doi/abs/10.1063/1.555664",
+    ),
+    referenceType = "experiment",
+    shortDesc = u""" """,
+    longDesc =
+u"""
+Bath gas: He
+Excitation technique:
```
Comment (alongd) [inline input/kinetics/libraries/Iodine-R_recombination/reactions.py]: missing info

File: input/kinetics/families/H_Abstraction/training/reactions.py [hunk 7]
```diff
@@ -64028,3 +64018,320 @@
 """,
 )
 
+entry(
+    index = 3079,
+    label = "I + CH4b => HI + CH3_p23",
```
[the remark this answers is not in the document]
Comment (sarahkha) [reply input/kinetics/families/H_Abstraction/training/reactions.py]: It should be in the library now.

File: input/thermo/groups/group.py [hunk 8]
```diff
@@ -46945,6 +46945,190 @@
 """,
 )
 
+entry(
```
[the remark this answers is not in the document]
Comment (alongd) [reply input/thermo/groups/group.py]: Would you like to "document" that Ct-CtI H298 is missing? You can do it in the commit message (after the header)

File: input/kinetics/libraries/Iodine-R_recombination/reactions.py (hunk 6 above)
Comment (sarahkha) [reply input/kinetics/libraries/Iodine-R_recombination/reactions.py]: @alongd 
Thanks a lot for your comments and your help!!

File: input/kinetics/families/H_Abstraction/groups.py [hunk 9]
```diff
@@ -7428,6 +7532,7 @@
             L4: N5dc_H
                 L5: N5dc/H/NonDeOO
         L3: HCl
+	    L3: HI
```
[the remark this answers is not in the document]
Comment (alongd) [reply input/kinetics/families/H_Abstraction/groups.py]: The indent should be identical to HCl above (same level)

File: input/kinetics/families/H_Abstraction/training/reactions.py [hunk 10]
```diff
@@ -63859,7 +63852,6 @@
     longDesc = 
 u"""
 Degeneracy not recalculated
-
```
Comment (alongd) [inline input/kinetics/families/H_Abstraction/training/reactions.py]: We still have the line deletions in the commit. Could you revert them?

File: input/kinetics/families/H_Abstraction/training/dictionary.txt [hunk 11]
```diff
@@ -4630,6 +4630,46 @@ multiplicity 2
...
+11    H u0 p0 c0 {2,S}
+12    H u0 p0 c0 {2,S}
+13    H u0 p0 c0 {2,S}
+14    H u0 p0 c0 {4,S}
+15    H u0 p0 c0 {6,S}
+16    H u0 p0 c0 {7,S}
+17    H u0 p0 c0 {8,S}
+18    H u0 p0 c0 {5,S}
+
+C8H9
```
Comment (alongd) [inline input/kinetics/families/H_Abstraction/training/dictionary.txt]: I think `C8H10` and `C8H9` shouldn't be part of this PR. Looks like they're not being used in the rates added here

File: input/kinetics/libraries/Iodine-R_recombination/reactions.py (hunk 1 above)
Comment (alongd) [reply input/kinetics/libraries/Iodine-R_recombination/reactions.py]: I think this comment wasn't addressed:
The units are for a 2nd-order reaction, but this looks like a 3rd-order one.
Same comment for the entry below

File: input/kinetics/libraries/Iodine-R_recombination/reactions.py [hunk 12]
```diff
@@ -0,0 +1,211 @@
...
+    u"""
+    Bath gas: Kr (100 Torr)
+    Estimated by Zach et al. by analogy to I + C2H5 -> C2H5I measured at 298 K, 100 Torr Kr bath gas and increased by 5 x to match MBMS experiments.
+    Supplementary Information (Table S2) for Buras, Z. J. et al. Phenyl Radical + Propene: A Prototypical Reaction Surface for Aromatic-Catalyzed 1,2Hydrogen-Migration and Subsequent Resonance Stabilized Radical Formation
+    """,
+)
+
+entry(
+    index = 7,
+    label = "C6H5CH2CHCH3 + I <=> C6H5CH2CHCH3I",
```
Comment (alongd) [inline input/kinetics/libraries/Iodine-R_recombination/reactions.py]: Looks to me like not all all species were defined in the dictionary

File: input/thermo/groups/group.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (alongd) [reply input/thermo/groups/group.py]: Was this comment addressed?

File: input/thermo/groups/group.py [hunk 13]
```diff
@@ -46945,6 +46945,190 @@
...
+    thermo = ThermoData(
+        Tdata = ([300,400,500,600,800,1000],'K'),
+        Cpdata = ([0.0,0.0,0.0,0.0,0.0,0.0],'cal/(mol*K)'),
+        H298 = (0.0,'kcal/mol'),
+        S298 = (0.0,'cal/(mol*K)'),
+    ),
+    shortDesc = u"""Dummy Iodine group""",
+    longDesc =
+u"""
+Dummy group for singly-bonded iodine. Benson groups for chloroalkanes already account for I in the Carbon-centered groups.
```
Comment (alongd) [inline input/thermo/groups/group.py]: Could you add a line break after `singly-bonded iodine.` so the sentence isn't visually too long?

File: input/thermo/libraries/iodinated_Hydrocarbons.py [hunk 14]
```diff
@@ -0,0 +1,388 @@
...
+    label = "I2",
+    molecule =
+"""
+1 I u0 p3 c0 {2,S}
+2 I u0 p3 c0 {1,S}
+""",
+    thermo=ThermoData(
+        Tdata=([500, 600, 800, 1000, 1500, 2000], 'K'),
+        Cpdata=([37.48, 37.60, 37.79, 38.06, 39.54, 42.86], 'J/(mol*K)'),
+        H298=(62.42, 'kJ/mol'),
```
Comment (alongd) [inline input/thermo/libraries/iodinated_Hydrocarbons.py]: Shouldn't H298 of I2 be zero by definition?

File: input/thermo/libraries/iodinated_Hydrocarbons.py [hunk 15]
```diff
@@ -0,0 +1,388 @@
...
+        S298=(260.687, 'J/(mol*K)'),
+    ),
+    shortDesc=u"""JPL Evaluation 18 + webbook.nist.gov""",
+    longDesc=
+    u"""
+Burkholder,J. B. et al.Chemical Kinetics and Photochemical Data for Use in Atmospheric Studies; Evaluation No. 18, JPL Publication 15–10; Jet Propulsion Laboratory: Pasadena, CA, 2015
```
Comment (alongd) [inline input/thermo/libraries/iodinated_Hydrocarbons.py]: Can you take a look at all the lines which are too long and add line breaks wherever you think is relevant?

File: input/kinetics/families/H_Abstraction/groups.py (hunk 9 above)
Comment (alongd) [reply input/kinetics/families/H_Abstraction/groups.py]: Was this comment addressed?

File: input/kinetics/libraries/Iodine-R_recombination/dictionary.txt [hunk 16]
```diff
@@ -0,0 +1,159 @@
+I
+multiplicity 2
+1 * I u1 p3 c0
```
Comment (alongd) [inline input/kinetics/libraries/Iodine-R_recombination/dictionary.txt]: We don't need to label atoms in libraries (i.e., with `*`).
Could you remove all labels?

File: input/kinetics/libraries/Iodine-R_recombination/reactions.py [hunk 17]
```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env python
+# encoding: utf-8
+
+name = "R_Recombination reactions parameters for iodine-containing species"
+shortDesc = u""" 
+(see Table S2 of Supplementary material of Buras et al.)
```
Comment (alongd) [inline input/kinetics/libraries/Iodine-R_recombination/reactions.py]: Could you add the full citation and doi?

File: input/kinetics/libraries/Iodine-R_recombination/reactions.py [hunk 18]
```diff
@@ -0,0 +1,211 @@
...
+        T0 = (1, 'K'),
+        Tmin = (290, 'K'),
+        Tmax = (1250, 'K'),
+    ),
+    reference = Article(
+        authors = ["DL Baulch, J Duxbury, SJ Grant, DC Montague"],
+        title = u'Evaluated kinetic data for high temperature reactions. Volume 4 Homogeneous gas phase reactions of halogen- and cyanide- containing species',
+        journal = "Journal of Physical and Chemical Reference Data",
+        volume = "10",
+        pages = """ """,
```
Comment (alongd) [inline input/kinetics/libraries/Iodine-R_recombination/reactions.py]: could you add relevant pages, or delete this line?

File: input/kinetics/families/H_Abstraction/training/reactions.py [hunk 19]
```diff
@@ -64121,3 +64121,286 @@
...
+Taken from entry: I_r3 + CH4_r12 => HI_p23 + CH3_p1
+Arrhenius parameters calculated at the CCSD(T)/ANO-RCC-L(I)//MP2/cc-pVTZ level of theory
+K.Meciarova et al., Chem. Phys. Lett., 517,149-154, 2011
+https://doi.org/10.1016/j.cplett.2011.10.029
+""",
+)
+
+entry(
+    index= 3086,
+    label = "HI + CH3_p23 <=> CH4b + I",
```
Comment (alongd) [inline input/kinetics/families/H_Abstraction/training/reactions.py]: I think the labels here are wrong. The reacting radical should be labelled with `*3`, but here it's `CH3_p23` (i.e., with `*2` and `*3` like a product). Same for `CH4b`, please find the appropriate species labels with the correct labelling from the dictionary of this family.

File: input/kinetics/families/H_Abstraction/training/reactions.py [hunk 20]
```diff
@@ -64121,3 +64121,286 @@
...
+Taken from entry:  I_r3 + H2_r12 => HI_p23 + H_p1
+Arrhenius parameters calculated at the CCSD(T)/cc-pVQZ//MP2/cc-pVTZ level of theory
+S.Canneaux, B.Xerri, F.Louis and L.Cantrel, J. Phys. Chem. A, 114, 9270–9288, 2010,
+https://DOI: 10.1021/jp104163t
+""",
+)
+
+entry(
+    index = 3088,
+    label = "HI + H <=> H2 + I",
```
Comment (alongd) [inline input/kinetics/families/H_Abstraction/training/reactions.py]: `H2` here should be `H2_p`. Could you take a second look at all labels in these reactions and correct accordingly?

File: input/thermo/libraries/iodinated_Hydrocarbons.py (hunk 14 above)
Comment (sarahkha) [reply input/thermo/libraries/iodinated_Hydrocarbons.py]: @alongd for I2(s) yes but here it is I2(g). The standard state for iodine (I2) is solid not gas.

File: input/kinetics/families/H_Abstraction/groups.py [hunk 21]
```diff
@@ -7461,7 +7461,7 @@
             L4: N5dc_H
                 L5: N5dc/H/NonDeOO
         L3: HCl
-	    L3: HI
```
Comment (alongd) [inline input/kinetics/families/H_Abstraction/groups.py]: Great! Feel free to squash this commit

File: input/thermo/groups/group.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (alongd) [reply input/thermo/groups/group.py]: Where did you decide to document it? (I don't see it in the commit message)

File: input/thermo/groups/group.py [hunk 22]
```diff
@@ -46945,6 +46945,190 @@
...
+4   I1s u0 {1,S}
+5   H   u0 {1,S}
+""",
+    thermo = ThermoData(
+        Tdata = ([300,500,600],'K'),
+        Cpdata = ([12.2, 16.4, 17.0],'cal/(mol*K)'),
+        H298 = (26.0,'kcal/mol'),
+        S298 = (54.6,'cal/(mol*K)'),
+    ),
+    shortDesc = u"""C-(I)2(C)(H) BENSON values in ()""",
```
Comment (alongd) [inline input/thermo/groups/group.py]: What is the `in ()`? Do we need it, or can we delete it?

File: input/thermo/groups/group.py (hunk 13 above)
Comment (alongd) [reply input/thermo/groups/group.py]: Was this change pushed? I still see a long line

File: input/kinetics/families/H_Abstraction/training/reactions.py (hunk 19 above)
Comment (alongd) [reply input/kinetics/families/H_Abstraction/training/reactions.py]: I still see `CH3_p23` as a reactant in `HI + CH3_p23 <=> CH4b + I` but as a product in `I + CH4b <=> HI + CH3_p23`. From our discussion I understood this was already fixed. Can you check whether the fixes got pushed?

**Comment by alongd** [comment]
The tests on this branch fail with `Species HI in kinetics depository H_Abstraction/training is missing from its dictionary`. Can you take a look?

File: input/kinetics/families/H_Abstraction/training/dictionary.txt [hunk 23]
```diff
@@ -5315,3 +5378,16 @@ HNO3_p
 3 O u0 p3 c-1 {2,S}
 4 O u0 p2 c0 {2,D}
 5 *1 H u0 p0 c0 {1,S}
+
+I
+multiplicity 2
+1 *3 I u1 p3 c0
+
+HI
```
[the remark this answers is not in the document]
Comment (sarahkha) [reply input/kinetics/families/H_Abstraction/training/dictionary.txt]: @alongd, HI is by default HI_r12 (see labels *1, *2), and HIp is HI_23 (please see below).

[the remark this answers is not in the document]
Comment (alongd) [reply input/kinetics/families/H_Abstraction/training/dictionary.txt]: I agree. Could you rename them to the non-official convention we use in this family? Please rename HI to HI_r12 and HIp to HI_p23 (e.g., the names chosen for HIp and C6H6_p23 don't follow the  same convention although they have the same labels)

**Comment by alongd** [comment]
@sarahkha, try rebasing this branch, I think the tests will pass now

File: input/kinetics/families/H_Abstraction/training/dictionary.txt [hunk 24]
```diff
@@ -5015,9 +5033,9 @@ multiplicity 2
 
 HO2
 multiplicity 2
-1 O u0 p2 c0 {2,S} {3,S}
-2 O u1 p2 c0 {1,S}
-3 H u0 p0 c0 {1,S}
+1 *1 O u0 p2 c0 {2,S} {3,S}
```
[the remark this answers is not in the document]
Comment (alongd) [reply input/kinetics/families/H_Abstraction/training/dictionary.txt]: @sarahkha, was this comment addressed?

**Comment by alongd** [comment]
@sarahkha, I see we already have some conflicts for rebasing this branch. Let me know once this branch is updated once again, and we'll get this PR in to avoid additional rebasing and conflict solving.

**Comment by alongd** [comment]
@sarahkha, the commits overall look OK, but I have a problem with the changes to the H abstraction groups and the thermo groups.py files. It seems like there was a merge conflict, and instead of solving it directly, these files were replaces altogether. For example, for thermo groups.py I see "49,302 additions, 51,549 deletions". Please make sure that this PR only includes the relevant specific additions.

File: input/kinetics/families/H_Abstraction/training/reactions.py [hunk 25]
```diff
@@ -64240,3 +64240,286 @@
...
+        Ea = (135.9, 'kJ/mol'),
+        T0 = (1, 'K'),
+        Tmin = (250, 'K'),
+        Tmax = (2500, 'K'),
+    ),
+    rank = 3,
+    shortDesc = u"""Training reaction from kinetics library: Iodine-H_abstraction""",
+    longDesc =
+u"""
+Taken from entry: I_r3 + CH4_r12 => HI_p23 + CH3_p1
```
Comment (alongd) [inline input/kinetics/families/H_Abstraction/training/reactions.py]: Should this line represent the reaction label in the reference? Looks like it's the RMG label (same comment for all entries below)

**Comment by sarahkha** [comment]
@alongd  Thank you so much!!!! 👍 :-)  :-) :-)
